### ISSUE TITLE
replay-supplement output zip files should unzip to a top-level directory rather than the current working directory

### ISSUE THREAD

**Issue by gregcaporaso** [qiime2/qiime2#731]
Currently when unzipping a reproducibility supplement, all files are extracted to the current working directory.  This should be updated so that the files are extracted to a directory with the same base name as the zip file. 

Current behavior:

```bash
(qiime2-amplicon-2023.9) ➜  q2results qiime tools replay-supplement \
  --in-fp . \
  --recurse \
  --out-fp ./reproducibility-supplement.zip
parsing unweighted_unifrac_emperor.qzv
parsing alpha-rarefaction.qzv
metadata written to recorded_metadata/
The <class 'q2cli.core.usage.ReplayCLIUsage'> replay script was written to cli_replay.sh.
metadata written to recorded_metadata/
The <class 'qiime2.core.archive.provenance_lib.usage_drivers.ReplayPythonUsage'> replay script was written to python3_replay.py.
The citations bibtex file was written to citations.bib.
The reproducibility package was written to /Users/gregcaporaso/temp/q2results/reproducibility-supplement.zip.

(qiime2-amplicon-2023.9) ➜  q2results ls
alpha-rarefaction.qzv          reproducibility-supplement.zip unweighted_unifrac_emperor.qzv

(qiime2-amplicon-2023.9) ➜  q2results unzip reproducibility-supplement.zip
Archive:  reproducibility-supplement.zip
   creating: recorded_metadata/
  inflating: cli_replay.sh
  inflating: citations.bib
  inflating: python3_replay.py
   creating: recorded_metadata/demux_emp_single_0/
   creating: recorded_metadata/diversity_alpha_rarefaction_0/
   creating: recorded_metadata/diversity_core_metrics_phylogenetic_0/
  inflating: recorded_metadata/diversity_core_metrics_phylogenetic_0/metadata_0.tsv
  inflating: recorded_metadata/diversity_alpha_rarefaction_0/metadata_1.tsv
  inflating: recorded_metadata/demux_emp_single_0/barcodes_0.tsv

(qiime2-amplicon-2023.9) ➜  q2results ls
alpha-rarefaction.qzv          cli_replay.sh                  recorded_metadata              unweighted_unifrac_emperor.qzv
citations.bib                  python3_replay.py              reproducibility-supplement.zip
```

It would help users stay organized and understand what files were created if that unzip command resulted in a top-level `reproducibility-supplement` directory with all of the files under that.

**Comment by colinvwood** [comment]
note: change "The <class 'q2cli.core.usage.ReplayCLIUsage'>..." to "The cli..." and "The <class 'qiime2.core.archive.provenance_lib.usage_drivers.ReplayPythonUsage'>" to "The python..."

**Comment by gregcaporaso** [comment]
> note: change "The <class 'q2cli.core.usage.ReplayCLIUsage'>..." to "The cli..." and "The <class
> 'qiime2.core.archive.provenance_lib.usage_drivers.ReplayPythonUsage'>" to "The python..."

Is there a description attribute on the `UsageDriver` class? If not, it's worth adding one so we have an attribute to access for this information, and it can fall back to the class descriptor (ie., current behavior) if that attribute isn't defined. 

cc @ebolyen

**Comment by colinvwood** [comment]
The `Usage` class doesn't have a metaclass to put `__str__` on and then you'd probably have to have two different metaclasses for the cli usage and the python usage classes or do some weird inheritance so I think I'll just change the print statement.

**Comment by colinvwood** [comment]
I guess we could just have a class attribute for the description like `script_type = 'python/cli'`, is that what you were thinking?

**Comment by colinvwood** [comment]
closed by #732
